### DISCUSSION TITLE
Encountered an Invalid block size error while using client.download()

### DISCUSSION THREAD

**Discussion by M1chaelWang** [gijzelaerr/python-snap7#539] [Q&A] [answered]
`download_data = bytearray(40)
err_code = client.download(download_data, 3)`
When running this line of code, the program threw an "Invalid block size" error and I can't figure why.
Please I need help. THX!!!  :)

**Comment by gijzelaerr** [comment, marked as answer]
Closing as stale. The `download()` function expects properly formatted S7 block data with valid block headers, not arbitrary bytearrays.
